### ISSUE TITLE
Large numbers of notifications for a single post don't appear to group correctly, and are split across multiple grouped notifications.

### ISSUE THREAD

**Issue by HeyItsLollie** [bluesky-social/social-app#1974]
**Describe the bug**

One of my posts has unexpectedly reached outside my usual circle, and is now exploding in notifications. The issue is that Liked and Reposted notifications for this single post are split across several unique groups of notifications, despite these Likes/Reposts being received within the same time period.

![firefox - 2023-11-22 - 11-40-34](https://github.com/bluesky-social/social-app/assets/30498581/5ad73db6-47fa-4748-a9c9-5a86bedaaed0)

This is only a sliver of the notifications that I'm wading through - there is a new group of Liked notifications for every 20-ish Likes received, and the post has 580+ Likes at this point.

**Steps to reproduce the behavior**

1. Make an unexpectedly banger post
2. Watch the notifications roll in
3. _Suffer_

**Expected behavior**

A post should only ever have one grouped notification for Likes and Reposts respectively, collected over a larger period of time.

That period of time should be variable based on the time that has elapsed between the last two previous instances of an action, and that time period should be very generous, eg: 24 hours between the last two Likes or Reposts added to a grouped notification, which should reset every time another notification is added to its respective group.

Example 1: If a post receives 99 Likes, but 24 hours pass without receiving any additional Likes for that specific post, that group of 99 Liked notifications should be frozen as a single grouped notification. Any additional Likes should then be collected in a new grouped notification.

Example 2: If a post receives 99 Likes, and 1 additional Like is added to that post within a 24-hour period, the user should only see a single grouped notification for 100 Likes for that post. The grouped notification count for a post should continue to grow, as long as the time between a post's last two notifications is less than 24 hours.
ie: A post could've been made _days_ ago, but as long as a post's previous two Likes or Reposts occur within 24 hours, that grouped notification should continue to grow until 24 hours pass without receiving a new notification of its respective type.

**Additional context**

I'm going to mute my post just so I can use my notifications again. The reason why I hadn't muted it initially is because I wanted to be able to see replies and quotes to this post as well.

**Comment by pfrazee** [comment]
Yeah unfortunately this would require server side changes. The reason it does this is because the client is getting pages of notifications and then grouping them
